### PR TITLE
CC-792 - Wrap ZK connection Xid back to zero rather than overflowing to negatve int32

### PR DESCRIPTION
**Pull request by kindkid**
Related: https://github.com/control-center/go-zookeeper/pull/1

### CODE REVIEW

File: Godeps [hunk 1]
```diff
@@ -73,8 +73,8 @@
 			"Rev": "43bd4e6d53ee382fd09bb6ede8b4029247444945"
 		},
 		{
-			"ImportPath": "github.com/samuel/go-zookeeper/zk",
-			"Rev": "4a222ca51b4c7ca6f2895e0e68af2e9254acfdcc"
+			"ImportPath": "github.com/zenoss/go-zookeeper/zk",
+			"Rev": "c34ab6262b97ef46e1ab13dd87cfca6de2042a15"
 		},
```
Comment (kindkid) [inline Godeps]: https://github.com/zenoss/go-zookeeper/compare/zenoss:4a222ca51b4c7ca6f2895e0e68af2e9254acfdcc...c34ab6262b97ef46e1ab13dd87cfca6de2042a15?expand=1

Comment (smousa) [inline Godeps]: Could you make it a uint instead?

**Comment by smousa** [comment]
I think we need to fork the repo in control-center

File: Godeps (hunk 1 above)
Comment (kindkid) [inline Godeps]: Unfortunately, no. There are a couple negative values that have meaning (-2 : ping)

**Comment by kindkid** [comment]
@smousa Oops! Forking to control-center would have made sense. I've already got a PR to samuel via the zenoss fork. Probably not worth closing and re-opening the PR now.

**Comment by kindkid** [comment]
Do not merge yet. Rebasing the go-zookeeper fix to exclude unrelated changes and add logging.

File: Godeps [hunk 2]
```diff
@@ -73,8 +73,8 @@
 			"Rev": "43bd4e6d53ee382fd09bb6ede8b4029247444945"
 		},
 		{
-			"ImportPath": "github.com/samuel/go-zookeeper/zk",
-			"Rev": "4a222ca51b4c7ca6f2895e0e68af2e9254acfdcc"
+			"ImportPath": "github.com/control-center/go-zookeeper/zk",
+			"Rev": "34c03e01311eb52816e48da2da97b5681bfbb1be"
```
Comment (kindkid) [inline Godeps]: https://github.com/control-center/go-zookeeper/compare/4a222ca51b4c7ca6f2895e0e68af2e9254acfdcc...34c03e01311eb52816e48da2da97b5681bfbb1be

**Pull request merged**